src/datasource.js

```diff
@@ -9,10 +9,19 @@ export class HawkularDatasource {
     this.type = instanceSettings.type;
     this.url = instanceSettings.url;
     this.name = instanceSettings.name;
-    this.tenant = instanceSettings.jsonData.tenant;
-    this.token = instanceSettings.jsonData.token;
     this.q = $q;
     this.backendSrv = backendSrv;
+    this.headers = {
+      'Content-Type': 'application/json',
+      'Hawkular-Tenant': instanceSettings.jsonData.tenant
+    };
+    if (typeof instanceSettings.basicAuth === 'string' && instanceSettings.basicAuth.length > 0) {
+      this.headers['Authorization'] = instanceSettings.basicAuth;
+    } else if (typeof instanceSettings.jsonData.username === 'string' && instanceSettings.jsonData.username.length > 0) {
+      this.headers['Authorization'] = 'Basic ' + window.btoa(instanceSettings.jsonData.username + ':' + instanceSettings.jsonData.password);
+    } else if (typeof instanceSettings.jsonData.token === 'string' && instanceSettings.jsonData.token.length > 0) {
+      this.headers['Authorization'] = 'Bearer ' + instanceSettings.jsonData.token;
```

Comment:
Could we add a notice in the datasource definition page to explain that token, if present, has higher priority than basic auth ?